tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,311 @@
+
+    The bootc disk creation requires running podman as root. The plugin will
+    automatically check if the current podman connection is rootless. If it is,
+    a podman machine will be spun up and used to build the bootc disk. The
+    podman machine can be configured with the following environment variables:
+
+    TMT_BOOTC_PODMAN_MACHINE_CPU='2'
+    TMT_BOOTC_PODMAN_MACHINE_MEM='2048'
+    TMT_BOOTC_PODMAN_MACHINE_DISK_SIZE='50'
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+    _rootless = True
+
+    def _get_id(self) -> str:
+        # FIXME: cast() - https://github.com/teemtee/tmt/issues/1372
+        parent = cast(tmt.steps.provision.Provision, self.parent)
+        assert parent.plan is not None
+        assert parent.plan.my_run is not None
+        assert parent.plan.my_run.unique_id is not None
+        return parent.plan.my_run.unique_id
+
+    def _expand_path(self, relative_path: str) -> str:
+        """ Expand the path to the full path relative to the current working dir """
+        if relative_path.startswith("/"):
+            return relative_path
+        return f"{os.getcwd()}/{relative_path}"
+
+    def _build_derived_image(self, base_image: str) -> str:
+        """ Build a "derived" container image from the base image with tmt dependencies added """
+        assert self.workdir is not None  # narrow type
+
+        simple_http_start_guest = \
+        """
+        python3 -m http.server {0} || python -m http.server {0} ||
+        /usr/libexec/platform-python -m http.server {0} || python2 -m SimpleHTTPServer {0} || python -m SimpleHTTPServer {0}
+        """.format(10022).replace('\n', ' ')
+
+        self._logger.debug("Building modified container image with necessary tmt packages/config")
+        containerfile_template = '''
+            FROM {{ base_image }}
+
+            RUN dnf -y install cloud-init rsync && \
+            ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants && \
+            rm /usr/local -rf && ln -sr /var/usrlocal /usr/local && mkdir -p /var/usrlocal/bin && \
+            dnf clean all && \
+            echo "{{ testcloud_guest }}" >> /opt/testcloud-guest.sh && \
+            chmod +x /opt/testcloud-guest.sh && \
+            echo "[Unit]" >> /etc/systemd/system/testcloud.service && \
+            echo "Description=Testcloud guest integration" >> /etc/systemd/system/testcloud.service && \
+            echo "After=cloud-init.service" >> /etc/systemd/system/testcloud.service && \
+            echo "[Service]" >> /etc/systemd/system/testcloud.service && \
+            echo "ExecStart=/bin/bash /opt/testcloud-guest.sh" >> /etc/systemd/system/testcloud.service && \
+            echo "[Install]" >> /etc/systemd/system/testcloud.service && \
+            echo "WantedBy=multi-user.target" >> /etc/systemd/system/testcloud.service && \
+            systemctl enable testcloud.service
+        '''
```

Comment:
I have a feeling this template (and the one with the HTTP server above) is starting to get out of hands - a complicated multiline string buried in a function. How about making it a module member, e.g. `CONTAINERFILE_TEMPLATE`, with some nice indentation of commands in it, maybe a comment or two? It's really hard for me to track what's happening there, it's very dense. Something like this would make it easier to decipher, at least for me:

```python
CONTAINER_TEMPLATE = """
FROM {{ base_image }}

RUN    dnf -y install cloud-init rsync \
    # Make sure cloud-init runs
    && ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants \
    # ...
    && rm /usr/local -rf \
    && ln -sr /var/usrlocal /usr/local \
    && mkdir -p /var/usrlocal/bin \
    && dnf clean all \
    # Workaround some testcloud problem??
    && echo "{{ testcloud_guest }}" >> /opt/testcloud-guest.sh \
    ...
```

Note I moved `&&` to the beginning of lines - I'm used to this way, but if you believe `&&` at the end is better, no problem. I'd just keep commands spread across multiple lines, with comments and better visibility into why they are there happening - as I reviewer, I can see some manipulation with `/usr/local`, but that's all... A hint or two would help future readers.